piptools/utils.py

```diff
@@ -630,7 +630,7 @@ def select_config_file(src_files: tuple[str, ...]) -> Path | None:
     # NOTE: input.
     working_directory = Path.cwd()
     src_files_as_paths = (
-        (working_directory / src_file).resolve() for src_file in src_files or (".",)
+        (working_directory / src_file).resolve() for src_file in src_files + (".",)
```

Comment:
Could you explain this? Obviously, there's some unspecified side effect in external logic, but I don't get it.